src/lib.rs

```diff
@@ -1,3 +1,130 @@
-//pub mod viam;
-mod espressif;
-pub use espressif::*;
+use micro_rdk::{
+    common::{
+        config::ConfigType,
+        registry::{ComponentRegistry, Dependency, RegistryError},
+        sensor::{
+            GenericReadingsResult, Readings, Sensor, SensorError, SensorResult, SensorT,
+            SensorType, TypedReadingsResult,
+        },
+        status::{Status, StatusError},
+    },
+    esp32::esp_idf_svc::hal::sys::esp,
+    DoCommand,
+};
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex},
+};
+
+// idf component re-exported from esp-idf-svc build system
+use micro_rdk::esp32::esp_idf_svc::sys::bme280::*;
+
+/// Global handle to I2C bus
+static mut I2C_BUS: i2c_bus_handle_t = std::ptr::null_mut();
+/// Global handle to BME280 device
+static mut BME280: bme280_handle_t = std::ptr::null_mut();
+const I2C_MODE_MASTER: u32 = 0x1;
+
+pub fn register_models(registry: &mut ComponentRegistry) -> Result<(), RegistryError> {
+    registry.register_sensor("bme280", &Bme280::from_config)
```

Comment:
as discussed, see if you can make this a triplet. not a blocker for merging.

Reply:
this is not possible without an adjustment to registration logic in micro-RDK